### DISCUSSION TITLE
(nginx) config for without sub domain

### DISCUSSION THREAD

**Discussion by mipimipi** [umputun/remark42#907] [Q&A] [answered]
I try to set up remark42 for my Hugo web site without sub domain and followed [this description](https://remark42.com/docs/1.6/subdomain/). Unfortunately, nginx complaints when I start the service:

    host not found in upstream "remark42" in /etc/nginx/sites-enabled/mankannsessen.de.conf:15

The reason is the line `proxy_pass ...` in the nginx conf (see below). If I comment that line nginx starts, but I get a 404 error when I enter `ansichten-eines-hobbykochs.de/remark42/web/embed.js` in my browser.

Do you have any idea what I have to change?

The container is named "remark42" (without quotation marks) and is up and running.

    $ docker ps              
    CONTAINER ID   IMAGE                     COMMAND                  CREATED       STATUS                 PORTS      NAMES
    7d0ff9b8210b   umputun/remark42:latest   "/init.sh /srv/remar…"   4 hours ago   Up 4 hours (healthy)   8080/tcp   remark42

That's my `docker-compose.yml`:

    version: '2'
    services:
        remark42:  
            image: umputun/remark42:latest
            container_name: remark42
            restart: always
            environment:
                - REMARK_URL=http://ansichten-eines-hobbykochs.de/remark42/
                - SECRET=********
                - SITE=mankannsessen-de
                - STORE_BOLT_PATH=/srv/var/db
                - BACKUP_PATH=/srv/var/backup
            volumes:
                - ./var:/srv/var

And that's my nginx conf:

        server {
                root /var/www/mankannsessen.de/public;
                
                index index.html;
                
                server_name ansichten-eines-hobbykochs.de www.ansichten-eines-hobbykochs.de;
                
                location / {
                        root /var/www/mankannsessen.de/public/html;
                        try_files $uri $uri/ =404;
                }
        
                location /remark42/ {
                        rewrite /remark42/(.*) /$1 break;
                        proxy_pass http://remark42:8080/;
                        proxy_set_header Host $http_host;
                        proxy_set_header X-Real-IP $remote_addr;
                        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                        proxy_set_header X-Forwarded-Proto $scheme;
                }
        }

**Comment by paskal** [comment, marked as answer]
It would work only if your Nginx will be running in the docker-compose as well, which might be not the case. If I am right, you should expose remark42 to the host:

```yaml
        ports:
            - "127.0.0.1:8080:8080"
```

Then change `proxy_pass http://remark42:8080/;` to `proxy_pass http://localhost:8080/;`, which should allow your Nginx to see the remark42 properly.

**Reply by mipimipi** [reply]
Thanks! And no, it also works it Nginx is running outside the docker-compose. It works with this config:
docker-compose.yml:

        version: '2'
        services:
            remark42:
                image: umputun/remark42:latest
                container_name: remark42
                restart: always
                ports:
                        - 8080:8080
                environment:
                        - REMARK_URL=http://ansichten-eines-hobbykochs.de/remark42
                        - SECRET=********
                        - SITE=mankannsessen-de
                        - STORE_BOLT_PATH=/srv/var/db
                        - BACKUP_PATH=/srv/var/backup
                volumes:
                        - ./var:/srv/var

nginx.conf:

        server {    
                root /var/www/mankannsessen.de/public;
                
                index index.html;
                
                server_name ansichten-eines-hobbykochs.de www.ansichten-eines-hobbykochs.de;
                
                location / {
                        root /var/www/mankannsessen.de/public/html;
                        try_files $uri $uri/ =404;
                }
                        
                location /remark42/ {
                        rewrite /remark42/(.*) /$1 break;
                        proxy_pass http://127.0.0.1:8080/;
                        proxy_set_header Host $http_host;
                        proxy_set_header X-Real-IP $remote_addr;
                        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                        proxy_set_header X-Forwarded-Proto $scheme;
                }
        }

Would be great if somebody could adjust the [docu](https://remark42.com/docs/1.6/subdomain/).

**Reply by umputun** [reply]
I'm not sure what to adjust. docs says "Both Nginx and Caddy configuration below relies on remark42 available on hostname remark42, which is achieved by having container_name: remark42 in docker-compose." and this is exactly the case.

`proxy_pass http://remark42:8080/;` works just fine if both nginx and remark42 defined in the same compose.  However, if containers started independently (not from the same compose) by default nginx won't know what "remark42" should be resolved to. It can be configured with docker network even in this case, but either way - this has very little to do with remark42 and just a part of the docker and docker-compose, decently documented on docker's site.

**Reply by mipimipi** [reply]
Yes, you are right. I didn't read this carefully. Thanks.

**Reply by paskal** [reply]
Having `port: - 8080:8080` might be not what you want, as that way it's exposed directly to the internet and not through your Nginx. I propose you to go with `127.0.0.1:8080:8080` to make it work only locally.
